python/pyspark/__init__.py

```diff
@@ -96,9 +96,11 @@ def keyword_only(func):
     """
     @wraps(func)
     def wrapper(*args, **kwargs):
+        # NOTE - this assumes we are wrapping a method and args[0] will be 'self'
         if len(args) > 1:
             raise TypeError("Method %s forces keyword arguments." % func.__name__)
         wrapper._input_kwargs = kwargs
```

Comment:
If the assumption is correct, should we always use 'self' to hold the kwargs? (remove this line and update all the fuctions that use `keyword_only`)?